### PR TITLE
MAINT: Warn that weights are not used in histogram bin edges calculation

### PR DESCRIPTION
**Pull request by domagalski**
Addressing issue #10682

### CODE REVIEW

File: numpy/lib/tests/test_histograms.py [hunk 1]
```diff
@@ -159,6 +160,15 @@ def test_weights(self):
             weights=[2, 1, 1, 1, 1, 1, 1, 1, 1], density=True)
         assert_almost_equal(a, [.2, .1, .1, .075])
 
+    def test_weights_warning(self):
+        a = np.array([1,2,3,4,5,6,7,8,9,10])
+        with warnings.catch_warnings(record=True) as w:
+            warnings.simplefilter("always")
+            hist, bin_edges = np.histogram(a,bins=10, weights=range(len(a)))
```
Comment (eric-wieser) [inline numpy/lib/tests/test_histograms.py]: Why should this warn?

Comment (charris) [reply numpy/lib/tests/test_histograms.py]: Want to use `assert_warns` in any case, I think. Any other tests where the warning is raised also need to suppress it.

**Comment by eric-wieser** [comment]
This warning should at most only be emitted for `bins="somestring"`, since it's meaningless in the other cases.

I'm not sure this is a suitable use of a warning at all though - there's no good way for a user to silence it.

**Comment by mhvk** [comment]
Indeed, the warning should only occur when the weights are in fact ignored - but I do think it is good for the user to realize this is happening! (#10682 is proof of that, I guess)

I discussed with @domagalski (student in my class; we'll work on things like pep8 offline) a bit what the appropriate warning would be: `UserWarning` seemed most logical, but I guess it depends in part on whether this would be something that happens a lot inside code that the user has no control over.

**Comment by eric-wieser** [comment]
I suppose we can encourage the user to write the following if they want to lock in the old behavior:
```
# compute the bins as if there were no weights
bins = np.histogram_bin_edges(x, bins='auto')
hist, _ = np.histograms(x, weights, bins=bins)
```
But that's slower, since it can't use the uniform bins optimization

**Comment by nayyarv** [comment]
I'm with @eric-wieser, this is certainly not a bug. I commented in more detail on the original issue #10682

Basically the weight argument should have no effect on calculating bin-edges when passed `bins=n`. as weights don't affect the range and number of bins, only the height of the histogram. 

This would affect any of the automated estimators (since the weight changes the distribution and size and thus the binwidth estimate), but trying to pass in weights with an automated estimator results in a TypeError, so that's already been covered.

Finally, the weights argument is literally only used to throw the above TypeError, so your warning would trigger every time weights was not None which would be annoying (for example it'd still trigger if you passed in an explicit bin array, which means we can't hold the function responsible for not using the weights arg). A good rule of thumb is to always test both success and failure - in this case, warnings raise when expected and warnings don't raise when not expected.

I think both the issue and this PR should be closed, or if the submitters feel strongly, update the documentation since everyone got a bit confused on how weights and edges interact.

**Comment by domagalski** [comment]
I think that's reasonable.

**Pull request closed without merging**